core/utils/bits.h

```diff
@@ -163,6 +163,26 @@ static inline void MaskBytes(uint8_t *buf, uint8_t const *mask,
   }
 }
 
+// TODO(melvinw): it would be nice if MaskTopN() supported blobs.
+// Returns a mask for the first `n` bits of an integral type `T`, i.e. starting
+// from the most significant bit toward the least significant. Returns a `T`
+// with all bits set if `n` is greater than the number of bits in `T`.
+// For example: MaskTopN<uint32_t>(9) will return 0xFF800000
+template <typename T, typename = std::enable_if<std::is_integral<T>::value>>
+static inline T SetBitsHigh(size_t n) {
+  return (n >= sizeof(T) * 8) ? ~T{0} : (T{1} << n) - 1;
+}
+
+// TODO(melvinw): it would be nice if MaskBottomN() supported blobs.
+// Returns a mask for the last `n` bits of an integral type `T`, i.e. starting
+// from the least significant bit toward the most signifant. Returns a `T` with
+// all bits set if `n` is greater than the number of bits in `T`.
+// For example: MaskBottomN<uint32_t>(9) will return 0x000001FF
```

Comment:
This comment is wrong, because `size_t` is unsigned: SetBitsLow<uint32_t>(31) works great (we get `~SetBitsHigh<uint32_t>(1)`), 32 works as well (32-32 = 0 = set no high bits = set all bits), but 33 means we do 32 - 33 = -1, convert to `size_t` and get some big number, big number > 32, `SetBitsHigh` returns 0xFFFFFFFF, and we get 0. This would need a similar (`n >= sizeof(T) * 8)` test.

(also, both comments need the new name 😺 )

Reply:
I was testing you :) Good catch!